In [1]:
import numpy as np

### Задача 1

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

#### Решение

Для расчета вероятностей воспользуемся формуллой Бернулли.

$P_n(X=k)=C_n^kp^kq^{n-k}$

Где $n=5$, $k$ это количество правильных ответов, а $p=\frac{1}{4}$

Сделаем расчет для всех $k$ от 0 до 5.

In [2]:
from scipy.special import comb as C

def bernulli(n, k, p):
    return C(n, k) * (p**k) * ((1-p)**(n-k))

p = 0.25
n = 5

P = [0] * 6

for k in [0, 1, 2, 3, 4, 5]:
    P[k] = round(bernulli(n, k, p), 3)

Для расчета $M(X)$ и $D(X)$ воспользуемся формулами математического ожидания и дисперсии для биномиального распределения.

$M(X)=np$

$D(X)=npq$

In [3]:
M = n * p
D = n * p * (1 - p)

#### Ответ

In [4]:
print(P)

[0.237, 0.396, 0.264, 0.088, 0.015, 0.001]


Пусть $X$ — количество правильных ответов. Тогда ее закон распределения выглядит так:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X=x)$</th>
        <td>0.237</td>
        <td>0.396</td>
        <td>0.264</td>
        <td>0.088</td>
        <td>0.015</td>
        <td>0.001</td>
    </tr>
</table>

In [5]:
print(f'M(X)={M}, D(X)={D}')

M(X)=1.25, D(X)=0.9375


### Задача 2

Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

#### Решение

Число попаданий - случайная величина. Ее закон распределения можно получить при помощи формулы Пуассона, т.к. вероятность события низкая,а число событий достаточно велико. Расчитаем значения вероятности, затем найдем сумму вероятностей для $x$ лежащего от 5 до 10 включительно.

In [6]:
p = 0.01
n = 200

def puasson(n, m, p):
    lambda_ = n * p
    return (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

P_sum_puasson = 0
for i in range(5, 11):
    P_sum_puasson += puasson(n, i, p)

#### Ответ

In [7]:
print(f'Вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10 равна '
      f'{P_sum_puasson}')

Вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10 равна 0.052644709119342686


### Задача 3

В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

Подсказка. Как посчитать P(X >= k) для величины X, имеющей распределение Пуассона? Здесь предстоит немножко покодить.

#### Решение

Выразим необходимые формулы

$P(X \geqslant k)=1-P(X<k)=1-\sum\limits_{k=0}^{k-1}P(X=k)$

$P(X=k)=\frac{λ^k}{k!}e^{−λ}=\frac{(np)^k}{k!}e^{−np}$

Посчитаем вероятность $P(X \geqslant k)$ для $k=10$, и для различных $n$ (от 10) пока вероятность не станеть $\geqslant$ 0.9

In [8]:
p = 0.01
k = 10
n = 10

def P_ge(n, k, p):
    sum_ = 0
    for k_ in range(0, k):  # до k-1+1
        sum_ += puasson(n, k_, p)
    return 1 - sum_

while P_ge(n, k, p) < 0.9:
    n += 1

#### Ответ

In [10]:
print(f'Нужно сделать {n} выстрелов, чтобы вероятность того что хотя бы {k} выстрелов попадут в цель была не менее 0.9.\n'
      f'Вероятность в таком случае равна {P_ge(n, k, p)}')

Нужно сделать 1421 выстрелов, чтобы вероятность того что хотя бы 10 выстрелов попадут в цель была не менее 0.9.
Вероятность в таком случае равна 0.9001760722984115
